syntax = "proto3";

package dofus_market.grpc_market;

import "google/protobuf/empty.proto";

service IngredientController {
    rpc Create(IngredientRequest) returns (IngredientResponse) {}
    rpc Destroy(IngredientDestroyRequest) returns (google.protobuf.Empty) {}
    rpc List(IngredientListRequest) returns (IngredientListResponse) {}
    rpc PartialUpdate(IngredientPartialUpdateRequest) returns (IngredientResponse) {}
    rpc Retrieve(IngredientRetrieveRequest) returns (IngredientResponse) {}
    rpc Update(IngredientRequest) returns (IngredientResponse) {}
}

service RecetteController {
    rpc Create(RecetteRequest) returns (RecetteResponse) {}
    rpc Destroy(RecetteDestroyRequest) returns (google.protobuf.Empty) {}
    rpc List(RecetteListRequest) returns (RecetteListResponse) {}
    rpc PartialUpdate(RecettePartialUpdateRequest) returns (RecetteResponse) {}
    rpc Retrieve(RecetteRetrieveRequest) returns (RecetteResponse) {}
    rpc Update(RecetteRequest) returns (RecetteResponse) {}
}

service RuneController {
    rpc Create(RuneRequest) returns (RuneResponse) {}
    rpc Destroy(RuneDestroyRequest) returns (google.protobuf.Empty) {}
    rpc List(RuneListRequest) returns (RuneListResponse) {}
    rpc PartialUpdate(RunePartialUpdateRequest) returns (RuneResponse) {}
    rpc Retrieve(RuneRetrieveRequest) returns (RuneResponse) {}
    rpc Update(RuneRequest) returns (RuneResponse) {}
}

message IngredientDestroyRequest {
    string name = 1;
}

message IngredientListRequest {
}

message IngredientListResponse {
    repeated IngredientResponse results = 1;
}

message IngredientPartialUpdateRequest {
    string name = 1;
    optional int32 price = 3;
    repeated string partial_update_fields = 4;
}

message IngredientRequest {
    string name = 1;
    optional int32 price = 2;
}

message IngredientResponse {
    string name = 1;
    optional int32 price = 2;
}

message IngredientRetrieveRequest {
    string name = 1;
}

message RecetteDestroyRequest {
    string ingredient = 1;
}

message RecetteListRequest {
}

message RecetteListResponse {
    repeated RecetteResponse results = 1;
}

message RecettePartialUpdateRequest {
    string ingredient = 1;
    repeated string partial_update_fields = 2;
    optional int32 level = 3;
    optional string metier = 4;
    repeated int64 ingredients = 5;
}

message RecetteRequest {
    string ingredient = 1;
    optional int32 level = 2;
    optional string metier = 3;
    repeated int64 ingredients = 4;
}

message RecetteResponse {
    string ingredient = 1;
    optional int32 level = 2;
    optional string metier = 3;
    repeated int64 ingredients = 4;
}

message RecetteRetrieveRequest {
    string ingredient = 1;
}

message RuneDestroyRequest {
    string name = 1;
}

message RuneListRequest {
}

message RuneListResponse {
    repeated RuneResponse results = 1;
}

message RunePartialUpdateRequest {
    string name = 1;
    optional int32 prix_ra = 3;
    optional int32 prix_pa = 4;
    optional int32 prix_ba = 5;
    repeated string partial_update_fields = 6;
}

message RuneRequest {
    string name = 1;
    optional int32 prix_ra = 2;
    optional int32 prix_pa = 3;
    optional int32 prix_ba = 4;
}

message RuneResponse {
    string name = 1;
    optional int32 prix_ra = 2;
    optional int32 prix_pa = 3;
    optional int32 prix_ba = 4;
}

message RuneRetrieveRequest {
    string name = 1;
}

